### ISSUE TITLE
Code fails here when I run jmReport task with unknown property 'enableReports' error

### ISSUE THREAD

**Issue by PushpanathanPerumal** [jmeter-gradle-plugin/jmeter-gradle-plugin#103]
https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin/blob/af6b5ced43f72ef4543a643c2dc922bd6ce7a367/src/main/groovy/net/foragerr/jmeter/gradle/plugins/TaskJMReports.groovy#L57

Error message:

Could not get unknown property 'enableReports' for object of type net.foragerr.jmeter.gradle.plugins.JMPluginExtension.

I think this is due to refactoring in jmInit task where these properties are removed. 

What should we do if we want to generate a single combined HTML report of all JMXs test results ?

**Comment by foragerr** [comment]
What version of the plugin?

**Comment by PushpanathanPerumal** [comment]
I am using 1.0.8-3.3 . All our jmeter tests are written in 3.3 version so using this one. As I looked at jmInit task code, I see that you removed "enableReports" property for some reason. But it is still being referred in jmReport task in latest version.

**Comment by foragerr** [comment]
Sorry about that. Removed enableReports with intent to add support for jmeter's built in reports that were introduced in 3.0. Thanks for the bug report, will get around to fixing it.

It is currently not possible to combine several results into a single report.

**Comment by markserrano915** [comment]
I get the same issue with `1.0.10-3.3` as well.  

` Could not set unknown property 'enableExtendedReports' for object of type net.foragerr.jmeter.gradle.plugins.JMPluginExtension.`

I have to remove the `enableExtendedReports` from the build.gradle configuration. 

I am able to run jmRun and jmGui but the jmReport fails because of that error above.

Good work by the way with this plugin.

**Comment by atcol** [comment]
Note this error doesn't occur with version `1.0.7-2.13` of the plugin.

**Comment by cschabl** [comment]
You can work around this issue with Groovy meta programming in an afterEvaluate block in your Gradle script like this:

```
afterEvaluate {
    def jMeterExt = project.extensions.findByName("jmeter")
    jMeterExt.metaClass.enableReports = true
    jMeterExt.metaClass.enableExtendedReports = false
    jMeterExt.metaClass.reportTitle = null
    jMeterExt.metaClass.reportPostfix = ""
    jMeterExt.metaClass.reportXslt = "$project.projectDir/jmeter-results-detail-report_21.xsl"
}
```

You can get `jmeter-results-detail-report_21.xsl` from a JMeter distribution (folder extra) or `jmeter-gradle-plugin-1.0.7.zip`. Extendend reports don't work, because of some version conflict with XStream library.

**Comment by crtanasescu** [comment]
Hello. @cschabl I have used your afterEvaluate block in gradle but I get following error: `jmReport FAILED
There are no results file to create reports from
Can't transform result`   

There is nothing else that can be done to use this command again? @foragerr

**Comment by cschabl** [comment]
Hi @crtanasescu, this is probably due to missing or wrong configuration in the jmeter block.

I used the following configuration:

```
jmeter {
    csvLogFile = false
    resultsLog = new File(project.buildDir, "jmeter-report/loadtest-results.xml")
}
```

**Comment by crtanasescu** [comment]
Thank you, @cschabl. It works now to make an xml with results, but I cannot generate the html graphs. Can I do also this when running >gradle jmReport ?

To be more specific, I get following error: 
'jmReport 
SystemId Unknown; Line #2; Column #28; stylesheet requires attribute: version
JMeter env exists. No one should see this normally.'

'Failed to load /~/jmeter-report/loadtest-results.xml using XStream. Error was: com.thoughtworks.xstream.mapper.CannotResolveClassException: org.apache.jmeter.save.TestResultWrapper'

**Comment by cschabl** [comment]
@crtanasescu, as mentioned in my comment above, extended reports which contain the graphs still don't work with my work-around.

**Comment by vlsergeyatsbt** [comment]
do we have a solution for extended reports with 1.0.10-3.3 version?

**Comment by foragerr** [comment]
Are folks still interested in the "old" reports and extendedreports? 
the older reports are based on xml, xslt templates, and some [graphs that jp@gc plugins generate](https://jmeter-plugins.org/wiki/GraphsGeneratorListener/) 

Jmeter 3.0 [added support for inbuilt reporting](https://jmeter.apache.org/usermanual/generating-dashboard.html). I was planning to deprecate the older reports and extendedreports in favor of generating Jmeter's built in reports, but this issue sounds like some of you are still using the older functionality?

**Comment by kebernet** [comment]
@foragerr Yeah. You are just killing me with this one. We are nailed to 3.3 because of a JMeter bug (https://bz.apache.org/bugzilla/show_bug.cgi?id=62463)  that is not getting fixed, and no version of your 3.3 deploys works.

**Comment by kebernet** [comment]
Also, All the docs are just painfully out of date.

**Comment by mikhail-g** [comment]
@foragerr 

> Are folks still interested in the "old" reports and extendedreports?
> the older reports are based on xml, xslt templates, and some [graphs that jp@gc plugins generate](https://jmeter-plugins.org/wiki/GraphsGeneratorListener/)
> 
> Jmeter 3.0 [added support for inbuilt reporting](https://jmeter.apache.org/usermanual/generating-dashboard.html). I was planning to deprecate the older reports and extendedreports in favor of generating Jmeter's built in reports, but this issue sounds like some of you are still using the older functionality?

how to generate this new Jmeter 3.0 report using a gradle plugin? I have an [Can't transform result](https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin/issues/126) error on jmReport gradle task
